thunder/dynamo/splitter.py

```diff
@@ -142,11 +144,18 @@ def is_thunder_supported_partition(node: torch.fx.Node) -> bool:
 
     # Call compile on the split region/s.
     thunder_compiled_fns = []
+    example_input_metadatas = []
     submodule_to_compiled_fns = {}
     for node in split_gm.graph.nodes:
         node_name = node.name
         if is_thunder_supported_partition(node):
             graph_module = getattr(split_gm, node.name)
+            # Record the input tensor metadata of the current module based on the faketensor 'example_value' of the placeholder node
```

Comment:
No need to split anything now. In the future, for similar situations, it's better to spin off some of the changes from a larger feature to separate pull requests while continuing to work on the larger feature.
The change here of recording the example inputs metadata is useful in general, not just for this PR. It could have been a perfect candidate for the spinoff pull request.